src/main/resources/idprovider/idprovider.es6

```diff
@@ -0,0 +1,321 @@
+import cache from '../lib/login-util/cache';
+import { createAccessMap, createUser, getDefaultGroups } from "../lib/login";
+import { getDynamicsUser } from "../lib/dynamics/get-dynamics-user";
+
+const configLib = require('/lib/config');
+const contextLib = require('/lib/context');
+const oidcLib = require('/lib/oidc');
+const loginLib = require('/lib/login');
+const requestLib = require('/lib/request');
+const preconditions = require('/lib/preconditions');
+const authLib = require('/lib/xp/auth');
+const portalLib = require('/lib/xp/portal');
+
+const ACTIONS = {
+    AFTER_VERIFY: '1'
+}
+
+function redirectToAuthorizationEndpoint() {
+    log.debug('Handling 401 error...');
+
+    const idProviderConfig = configLib.getIdProviderConfig();
+    const redirectUri = generateRedirectUri();
+
+    const state = oidcLib.generateToken();
+    const nonce = oidcLib.generateToken();
+    const originalUrl = requestLib.getRequestUrl();
+    const context = {
+        state: state,
+        nonce: nonce,
+        originalUrl: originalUrl,
+        redirectUri: redirectUri
+    };
+    log.debug('Storing context: ' + JSON.stringify(context));
+    requestLib.storeContext(context);
+
+    const authorizationUrl = oidcLib.generateAuthorizationUrl({
+        authorizationUrl: idProviderConfig.authorizationUrl,
+        clientId: idProviderConfig.clientId,
+        redirectUri: redirectUri,
+        scopes: 'openid' + (idProviderConfig.scopes ? ' ' + idProviderConfig.scopes : ''),
+        state: state,
+        nonce: nonce
+    });
+    log.debug('Generated authorization URL: ' + authorizationUrl);
+
+    return {
+        redirect: authorizationUrl
+    };
+}
+
+function generateRedirectUri(params = {}) {
+    return portalLib.idProviderUrl({
+        idProviderKey: portalLib.getIdProviderKey(),
+        type: 'absolute',
+        params
+    });
+}
+
+/**
+ * Called when a user returns successfully from validation
+ * @param params
+ */
+function userVerified({ params }) {
+    log.info('User verified request received');
+    log.info(JSON.stringify(params, null, 4));
```

Comment:
antar det er meningen å logge dette for nå?

Reply:
Hm, akkurat den loggingen kan jeg fjerne nå.

Reply:
Fikset